packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -226,41 +291,100 @@ export class ValidatorApi implements IValidatorApi {
+    this.notWhileSyncing();
+
+    await Promise.all(
+      contributionAndProofs.map(async (contributionAndProof) => {
+        // TODO: Validate in batch
+        await validateSyncCommitteeGossipContributionAndProof(this.config, this.chain, this.db, {
+          contributionAndProof,
+          validSignature: false,
+        });
+        this.db.syncCommitteeContribution.add(contributionAndProof.message);
+        await this.network.gossip.publishContributionAndProof(contributionAndProof);
       })
     );
   }
 
   async prepareBeaconCommitteeSubnet(subscriptions: phase0.BeaconCommitteeSubscription[]): Promise<void> {
     this.notWhileSyncing();
 
-    this.network.prepareBeaconCommitteeSubnet(subscriptions);
+    this.network.prepareBeaconCommitteeSubnet(
+      subscriptions.map(({validatorIndex, slot, isAggregator, committeesAtSlot, committeeIndex}) => ({
+        validatorIndex: validatorIndex,
+        subnet: computeSubnetForCommitteesAtSlot(this.config, slot, committeesAtSlot, committeeIndex),
+        slot: slot,
+        isAggregator: isAggregator,
+      }))
+    );
 
     // TODO:
     // If the discovery mechanism isn't disabled, attempt to set up a peer discovery for the
     // required subnets.
   }
 
+  /**
+   * POST `/eth/v1/validator/sync_committee_subscriptions`
+   *
+   * Subscribe to a number of sync committee subnets.
+   * Sync committees are not present in phase0, but are required for Altair networks.
+   * Subscribing to sync committee subnets is an action performed by VC to enable network participation in Altair networks,
+   * and only required if the VC has an active validator in an active sync committee.
+   *
+   * https://github.com/ethereum/eth2.0-APIs/pull/136
+   */
+  async prepareSyncCommitteeSubnets(subscriptions: altair.SyncCommitteeSubscription[]): Promise<void> {
+    this.notWhileSyncing();
+
+    const state = await this.chain.getHeadStateAtCurrentEpoch();
+
+    // TODO: Cache this value
+    const SYNC_COMMITTEE_SUBNET_SIZE = Math.floor(this.config.params.SYNC_COMMITTEE_SIZE / SYNC_COMMITTEE_SUBNET_COUNT);
+
+    // A `validatorIndex` can be in multiple subnets, so compute the CommitteeSubscription with double for loop
+    const subs: CommitteeSubscription[] = [];
+    for (const sub of subscriptions) {
+      const committeeIndices = state.currSyncComitteeValidatorIndexMap.get(sub.validatorIndex);
+      if (committeeIndices) {
+        for (const committeeIndex of committeeIndices) {
+          const subnet = Math.floor(committeeIndex / SYNC_COMMITTEE_SUBNET_SIZE);
+          subs.push({
+            validatorIndex: sub.validatorIndex,
+            subnet: subnet,
+            // Subscribe until the end of `untilEpoch`: https://github.com/ethereum/eth2.0-APIs/pull/136#issuecomment-840315097
+            slot: computeStartSlotAtEpoch(this.config, sub.untilEpoch + 1),
+            isAggregator: true,
```

Comment:
similar to the other concern, in my understanding this is different from `prepareBeaconCommitteeSubnet` so we don't have to refactor to have `CommitteeSubscription` type

Reply:
See other replies